src/Engine/Manager/RenderManager.cpp

```diff
@@ -283,26 +314,36 @@ void RenderManager::Render(World& world, const glm::mat4& viewMatrix, const glm:
 
 }
 
-Component::Animation* RenderManager::CreateAnimation() {
-    return animations.Create();
+void RenderManager::UpdateAnimations(float deltaTime) {
+    // Update all enabled animation controllers.
+    for (Component::AnimationController* animationController : animationControllers.GetAll()) {
+        if (animationController->IsKilled() || !animationController->entity->enabled)
+            continue;
+    
+        animationController->UpdateAnimation(deltaTime);
+    }
 }
 
-Component::Animation* RenderManager::CreateAnimation(const Json::Value& node) {
-    Component::Animation* animation = animations.Create();
-    
-    // Load values from Json node.
-    std::string name = node.get("riggedModel", "").asString();
-    /// @todo Fix animation.
-    /*for (Geometry::Model* model : Hymn().models) {
-        if (model->name == name)
-            riggedModel = model;
-    }*/
+Component::AnimationController* RenderManager::CreateAnimation() {
+    return animationControllers.Create();
+}
+
+Component::AnimationController* RenderManager::CreateAnimation(const Json::Value& node) {
+    Component::AnimationController* animationController = animationControllers.Create();
     
-    return animation;
+    std::string skeletonName = node.get("skeleton", "").asString();
+    if (!skeletonName.empty())
+        animationController->skeleton =  Managers().resourceManager->CreateSkeleton(skeletonName);
+
+    std::string controllerName = node.get("animationController", "").asString();
+    if (!controllerName.empty())
+        animationController->controller =  Managers().resourceManager->CreateAnimationController(controllerName);
+
+    return animationController;
 }
 
-const std::vector<Component::Animation*>& RenderManager::GetAnimations() const {
-    return animations.GetAll();
+const std::vector<Component::AnimationController*>& RenderManager::GetAnimations() const {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetAnimations' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)